iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -243,158 +279,162 @@ TYPED_TEST(RelativePointer_test, AssignmentOperatorTests)
+    // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    EXPECT_EQ(rp1.getOffset(), reinterpret_cast<std::ptrdiff_t>(typedPtr));
+    EXPECT_EQ(rp1.getId(), 1U);
+
+    int offset = SHARED_MEMORY_SIZE / 2U;
+    // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
+    RelativePointer<TypeParam> rp2(addressAtOffset, iox::rp::BaseRelativePointer::id_t{1U});
     EXPECT_EQ(rp2.getOffset(), offset);
-    EXPECT_EQ(rp2.getId(), 1);
+    EXPECT_EQ(rp2.getId(), 1U);
     EXPECT_EQ(rp2.get(), addressAtOffset);
 }
 
-TYPED_TEST(RelativePointer_test, getOffset)
+TYPED_TEST(RelativePointer_test, GetOffsetReturnsCorrectOffset)
 {
     ::testing::Test::RecordProperty("TEST_ID", "0b493337-ee55-498a-9cac-8bb5741f72f0");
-    auto ptr = this->partitionPtr(0);
-    auto typedPtr = reinterpret_cast<TypeParam*>(ptr);
-
-    RelativePointer<TypeParam> rp1(typedPtr, 1);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
-    EXPECT_EQ(BaseRelativePointer::getOffset(1, ptr), 0);
-
-    int offset = SHARED_MEMORY_SIZE / 2;
-    auto addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
-    RelativePointer<TypeParam> rp2(addressAtOffset, 1);
-    EXPECT_EQ(BaseRelativePointer::getOffset(1, addressAtOffset), offset);
+    auto* ptr = this->partitionPtr(0U);
+    auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr));
+
+    RelativePointer<TypeParam> rp1(typedPtr, iox::rp::BaseRelativePointer::id_t{1U});
+    EXPECT_EQ(rp1.registerPtr(iox::rp::BaseRelativePointer::id_t{1U}, typedPtr), true);
+    EXPECT_EQ(BaseRelativePointer::getOffset(iox::rp::BaseRelativePointer::id_t{1U}, typedPtr), 0U);
+
+    int offset = SHARED_MEMORY_SIZE / 2U;
+    // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
+    RelativePointer<TypeParam> rp2(addressAtOffset, iox::rp::BaseRelativePointer::id_t{1U});
+    EXPECT_EQ(BaseRelativePointer::getOffset(iox::rp::BaseRelativePointer::id_t{1U}, addressAtOffset), offset);
 }
 
-TYPED_TEST(RelativePointer_test, getPtr)
+TYPED_TEST(RelativePointer_test, GetPtrReturnsAddressWithCorrectOffset)
 {
     ::testing::Test::RecordProperty("TEST_ID", "4fadf89f-69c0-4058-8995-a98e2e3334b2");
-    auto ptr = this->partitionPtr(0);
-    auto typedPtr = reinterpret_cast<TypeParam*>(ptr);
-    RelativePointer<TypeParam> rp1(typedPtr, 1);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
-    EXPECT_EQ(BaseRelativePointer::getPtr(1, 0), ptr);
+    auto* ptr = this->partitionPtr(0U);
+    auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(this->partitionPtr(0U)));
```

Comment:
please use reinterpret_cast

Reply:
`static_cast` has several advantages e.g. it can be used during compile-time, see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1521#discussion_r936463947).